src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -0,0 +1,164 @@
+                    message = string.Format(SR.ExpectedEndOfDigitNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedNextDigitComponentNotFound:
+                    message = string.Format(SR.ExpectedNextDigitComponentNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedNextDigitEValueNotFound:
+                    message = string.Format(SR.ExpectedNextDigitEValueNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedSeparaterAfterPropertyNameNotFound:
+                    message = string.Format(SR.ExpectedSeparaterAfterPropertyNameNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedStartOfPropertyNotFound:
+                    message = string.Format(SR.ExpectedStartOfPropertyNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedStartOfPropertyOrValueNotFound:
+                    message = SR.ExpectedStartOfPropertyOrValueNotFound;
+                    break;
+                case ExceptionResource.ExpectedStartOfValueNotFound:
+                    message = string.Format(SR.ExpectedStartOfValueNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedValueAfterPropertyNameNotFound:
+                    message = SR.ExpectedValueAfterPropertyNameNotFound;
+                    break;
+                case ExceptionResource.FoundInvalidCharacter:
+                    message = string.Format(SR.FoundInvalidCharacter, character);
+                    break;
+                case ExceptionResource.InvalidEndOfJson:
+                    message = string.Format(SR.InvalidEndOfJson, json.TokenType);
+                    break;
+                case ExceptionResource.ObjectDepthTooLarge:
+                    message = string.Format(SR.ObjectDepthTooLarge, json.CurrentDepth, json.CurrentState.MaxDepth);
+                    break;
+                case ExceptionResource.ObjectEndWithinArray:
+                    if (json.CurrentDepth <= 0)
+                        message = string.Format(SR.DepthMustBePositive, json.CurrentDepth);
+                    else
+                        message = string.Format(SR.ObjectEndWithinArray);
+                    break;
+                case ExceptionResource.Default:
+                    break;
+                case ExceptionResource.ExpectedFalse:
+                    message = string.Format(SR.ExpectedFalse, characters);
+                    break;
+                case ExceptionResource.ExpectedNull:
+                    message = string.Format(SR.ExpectedNull, characters);
+                    break;
+                case ExceptionResource.ExpectedTrue:
+                    message = string.Format(SR.ExpectedTrue, characters);
+                    break;
+                // This case is covered between ArrayEndWithinObject and ObjectEndWithinArray
```

Comment:
nit: dead code

Reply:
I don't want someone in the future do "Add missing cases" via VS refactoring on the switch type and add the case incorrectly, which is why I have the comment.

Reply:
I'd add the case label either to the (pending) default: Debug.Fail, or give it a dedicated Debug.Fail to report that the caller shouldn't have passed it (which could be shared with other "implied") values in the future.